### PR TITLE
Bug fixes for the trees used in the entire program

### PR DESCRIPTION
**Pull request by kRichard32**
Most features are now working, except for the data persistence features. Various bugs involving traversing trees and performing actions have been fixed. Action spaces and corner cases are still not implemented completely.

### CODE REVIEW

File: src/save/properties.txt [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-Pennsylvania Railroad,Railroad,200,0,25,50,100,200,0,100,0
+Pennsylvania Railroad,Railroad,200,0,25,50,100,200,0,100,0,0
 St. James Place,Orange,180,100,14,70,200,550,750,950,90,0
 Tennessee Avenue,Orange,180,100,14,70,200,550,750,950,90,0
 New York Avenue,Orange,200,100,16,80,220,600,800,1000,100,0
 Kentucky Avenue,Red,220,150,18,90,250,700,875,1050,110,0
 Indiana Avenue,Red,220,150,18,90,250,700,875,1050,110,0
 Illinois Avenue,Red,240,150,20,100,300,750,925,1100,120,0
-B&O Railroad,Railroad,200,0,25,50,100,200,0,100,0
+B&O Railroad,Railroad,200,0,25,50,100,200,0,100,0,0
```
Comment (benxu3) [inline src/save/properties.txt]: I will add another 0 between 200 and 100 at the end of the line to denote empty rentHotel value and remove the 0 from the end such that this parses correctly.

**Review by benxu3 — APPROVED** [review]
Looks good. A bit hard to understand the code changes without any JavaDoc, remember to implement those and Unit Tests once you get the chance. We can handle any edge cases that arise during testing in follow-up PRs.

**Review by harshkpatel — APPROVED** [review]
I think from a high level what you are trying to do with debugging involves using previous state to make a comparison with current state so the game doesn't run into issues during the game loop. The changes look acceptable for now with no real glaring error in convention or anything. I am also going to assume that you have tested the changes you've made on your own and have yet to formalize them, which can be dealt with in a later PR. As Ben said earlier, some of the stuff is still hard to understand without proper documentation, but just off variable names and changes you made, I can surmise that overall the changes look good and that we can handle any problems with future PRs and more refactoring.

**Comment by kRichard32** [comment]
Previous state is not used for debugging, its used for giving the player the option of a back button in specific menus.

**Comment by harshkpatel** [comment]
> Previous state is not used for debugging, its used for giving the player the option of a back button in specific menus.

That adds up, good design choice!

**Review by derrcodes — APPROVED** [review]
Everything seems good. The implementation is quite a lot so it was a little difficult to understand due to the lack of documentation but I'm sure this can be added later and its nto that big of a deal. Like Harsh, I will assume the implementation has been tested since its quite dense for me to actually see if there are any major issues. Otherwise everything seems good.
